src/azul/__init__.py

```diff
@@ -343,6 +343,34 @@ def qualified_resource_name(self, resource_name, suffix='', stage=None):
             stage = self.deployment_stage
         return f"{self.resource_prefix}-{resource_name}-{stage}{suffix}"
 
+    def unqualified_lambda_name(self, lambda_function: str) -> Tuple[str, str, Optional[str]]:
```

Comment:
The intention behind the `suffix` parameter was to support parsing of lambda function names. We might need to tweak the handling of suffix a bit:

```diff
Index: src/azul/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/__init__.py b/src/azul/__init__.py
--- a/src/azul/__init__.py	(revision 4b2cab8413e33844a4e2e70ecfa6c910c678d2d6)
+++ b/src/azul/__init__.py	(date 1616459128019)
@@ -345,7 +345,7 @@
 
     def unqualified_resource_name(self,
                                   qualified_resource_name: str,
-                                  suffix: str = ''
+                                  suffix: str = None
                                   ) -> Tuple[str, str]:
         """
         >>> config.unqualified_resource_name('azul-foo-dev')
@@ -364,10 +364,13 @@
         ...
         azul.RequirementError: ['azul', 'object', 'versions', 'dev']
         """
-        require(qualified_resource_name.endswith(suffix))
-        if len(suffix) > 0:
+        sep = '-'
+        if suffix is not None:
+            require(len(suffix) > 0, suffix)
+            suffix = sep + suffix
+            require(qualified_resource_name.endswith(suffix))
             qualified_resource_name = qualified_resource_name[:-len(suffix)]
-        components = qualified_resource_name.split('-')
+        components = qualified_resource_name.split(sep)
         require(len(components) == 3, components)
         prefix, resource_name, deployment_stage = components
         require(prefix == self.resource_prefix)
```

after which we should be able to use it for lambda function names:

```python
from azul import config
config.unqualified_resource_name('azul-indexer-dev-contribute_retry', suffix='contribute_retry')
('indexer', 'dev')
```